apps/nextjs-website/src/components/atoms/EmbedPageInfo/EmbedPageInfo.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Box,
+  Card,
+  CardActionArea,
+  CardContent,
+  Typography,
+} from '@mui/material';
+import { ButtonNaked } from '@/editorialComponents/Footer/components/ButtonNaked';
+import Link from 'next/link';
+import Image from 'next/image';
+import EmbedCaption from '@/components/atoms/EmbedCaption/EmbedCaption';
+
+const getProtocolAndHostFormUrl = (url: string): string => {
+  const urlObject = new URL(url);
+  return `${urlObject.protocol}//${urlObject.host}`;
+};
+
+const getFaviconUrl = (url: string, path: string): string => {
+  if (path.startsWith('http')) return path;
+  return `${getProtocolAndHostFormUrl(url)}${
+    path.startsWith('/') ? path : `/${path}`
+  }`;
+};
+
+export type EmbedPageInfoProps = {
+  url: string;
+  children?: React.ReactNode;
+};
+
+const EmbedPageInfo = ({ url, children }: EmbedPageInfoProps) => {
+  const [pageTitle, setPageTitle] = useState<string>('');
+  const [siteName, setSiteName] = useState<string>('');
+  const [favicon, setFavicon] = useState<string>('');
+
+  useEffect(() => {
+    async function fetchPageInfo() {
+      // Fetch the URL to get the page's HTML
+      const html = await fetch(url)
+        .then((res) => res.text())
+        .catch((error) => console.error('Error fetching data:', error));
```

Comment:
What about to render a fallback component instead to print an error?

```suggestion
```

Reply:
I agree to handle the case of fetch crashing by showing a fallback component (maybe some "this link is broken").

But I'd leave the catch there, because in case of a network error, fetch would throw an exception (all native js api does) and that needs to be catched and handled. 

If the catch stays there, html would be null in case of an error. So in the renderer we could add something like

```
if(!html) {
  return <FallbackComponent/>
}
```

Reply:
consider that html is null before fetching and on loading. what about using [swr](https://swr.vercel.app/docs/with-nextjs) or react-query for client driven fetches?

Reply:
https://github.com/pagopa/developer-portal/pull/315#discussion_r1365477346